File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> stack;
        for(int i = 0; i < s.size(); i++){
            char ch = s[i];
            if(ch == '(' || ch == '{' or ch == '['){stack.push(ch);}
            else{
                if(stack.empty() || (stack.top() == '(' && ch != ')') || (stack.top() == '{' && ch != '}') || (stack.top() == '[' && ch != ']')){return false;}
                stack.pop();
            }
        }
        return stack.empty();
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for char in s:
            if char in '({[':
                stack.append(char)
            else:
                if not stack:
                    return False
                begin = stack[-1]
                if (begin != '(' and char == ')' )or (begin != '[' and char == ']') or (begin != '{' and char == '}'):
                    return False
                else:
                    stack.pop()

        return not stack

                
